src/program_state.inl

```diff
@@ -895,6 +1057,16 @@ public:
         return it0->second;
     }
 
+    const std::size_t getKernattribute(std::string kernelName) {
+
+        auto it = get_kernattribute().find(kernelName);
+	if(it == get_kernattribute().end())
+	{
+	    hip_throw(std::runtime_error{"Undefined __global__ function."});
```

Comment:
How about adding the `kernelName` to the exception string thrown to help test and debug?